File: Strings/LargestOddNumberInString/largestOdd.go
package main

import (
    "fmt"
)

func largestOddNumber(num string) string {
    for i := len(num) - 1; i >= 0; i-- {
        digit := num[i] - '0'     // convert byte to int (ASCII trick)
        if digit%2 == 1 {
            return num[:i+1]      // return prefix
        }
    }
    return ""
}

func main() {
    fmt.Println(largestOddNumber("7542356")) // "754235"
    fmt.Println(largestOddNumber("4206"))    // ""
}

File: Strings/LargestOddNumberInString/largestOdd.ts
function largestODD(num:string):string{
    for(let i =num.length-1; i>= 0; i--){
         if (parseInt(num[i]) % 2 === 1) {   // convert char → number
            return num.substring(0, i + 1);
        }
    }
    return ""
}